### PR TITLE
 Convert  tests in "app/plugin_hooks_test.go" to use assert/require 

### PR DESCRIPTION
**Pull request by arshchimni**
#### Summary 
This pr is in reference to migrating the tests from "app/plugin_hooks_test.go" to use testify. 
The t.Fatal have been replaced with require and the Errorf have been replaced by apt assert package methhods.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12058

### CODE REVIEW

**Comment by arshchimni** [comment]
Seems like it is getting timed out during the test stage in jenkins

File: app/plugin_hooks_test.go [hunk 1]
```diff
@@ -678,11 +663,7 @@ func TestUserWillLogIn_Blocked(t *testing.T) {
 	defer th.TearDown()
 
 	err := th.App.UpdatePassword(th.BasicUser, "hunter2")
-
-	if err != nil {
-		t.Errorf("Error updating user password: %s", err)
-	}
-
+	assert.Empty(t, err, "Error updating user password: %s", err)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: As we are using `assert.Nil` for `nil` errors in other tests, can you please update this to use it for consistency?

File: app/plugin_hooks_test.go [hunk 2]
```diff
@@ -217,15 +215,14 @@ func TestHookMessageWillBePosted(t *testing.T) {
...
-		if retrievedPost, errSingle := th.App.Srv.Store.Post().GetSingle(post.Id); err != nil {
-			t.Fatal(errSingle)
-		} else {
-			assert.Equal(t, "message_fromplugin", retrievedPost.Message)
-		}
+		retrievedPost, errSingle := th.App.Srv.Store.Post().GetSingle(post.Id)
+		require.Nil(t, errSingle)
+
+		assert.Equal(t, "message_fromplugin", retrievedPost.Message)
+
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: nit: unnecessary empty line

File: app/plugin_hooks_test.go [hunk 3]
```diff
@@ -722,9 +701,7 @@ func TestUserWillLogInIn_Passed(t *testing.T) {
 
 	err := th.App.UpdatePassword(th.BasicUser, "hunter2")
 
-	if err != nil {
-		t.Errorf("Error updating user password: %s", err)
-	}
+	assert.Empty(t, err, "Error updating user password: %s", err)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: Same here

File: app/plugin_hooks_test.go [hunk 4]
```diff
@@ -754,13 +731,9 @@ func TestUserWillLogInIn_Passed(t *testing.T) {
 	w := httptest.NewRecorder()
 	session, err := th.App.DoLogin(w, r, th.BasicUser, "")
 
-	if err != nil {
-		t.Errorf("Expected nil, got %s", err)
-	}
+	assert.Empty(t, err, "Expected nil, got %s", err)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: Same here

File: app/plugin_hooks_test.go [hunk 5]
```diff
@@ -754,13 +731,9 @@ func TestUserWillLogInIn_Passed(t *testing.T) {
...
 
-	if session.UserId != th.BasicUser.Id {
-		t.Errorf("Expected %s, got %s", th.BasicUser.Id, session.UserId)
-	}
+	assert.Equal(t, session.UserId, th.BasicUser.Id)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: This assertion is missing the error message

File: app/plugin_hooks_test.go [hunk 6]
```diff
@@ -769,9 +742,7 @@ func TestUserHasLoggedIn(t *testing.T) {
 
 	err := th.App.UpdatePassword(th.BasicUser, "hunter2")
 
-	if err != nil {
-		t.Errorf("Error updating user password: %s", err)
-	}
+	assert.Empty(t, err, "Error updating user password: %s", err)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: Same `assert.Nil` change here

File: app/plugin_hooks_test.go [hunk 7]
```diff
@@ -802,17 +773,14 @@ func TestUserHasLoggedIn(t *testing.T) {
 	w := httptest.NewRecorder()
 	_, err = th.App.DoLogin(w, r, th.BasicUser, "")
 
-	if err != nil {
-		t.Errorf("Expected nil, got %s", err)
-	}
+	assert.Empty(t, err, "Expected nil, got %s", err)
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: Same here

File: app/plugin_hooks_test.go [hunk 8]
```diff
@@ -802,17 +773,14 @@ func TestUserHasLoggedIn(t *testing.T) {
...
 
 	time.Sleep(2 * time.Second)
 
 	user, _ := th.App.GetUser(th.BasicUser.Id)
 
-	if user.FirstName != "plugin-callback-success" {
-		t.Errorf("Expected firstname overwrite, got default")
-	}
+	assert.Equal(t, user.FirstName, "plugin-callback-success", "Expected firstname overwrite, got default")
+
```
Comment (mgdelacroix) [inline app/plugin_hooks_test.go]: nit: unnecessary empty line

File: app/plugin_hooks_test.go (hunk 5 above)
Comment (jespino) [reply app/plugin_hooks_test.go]: I don't think is needed, the `assert.Equal` is going to give you that exact information anyway.

Comment (mgdelacroix) [reply app/plugin_hooks_test.go]: Totally true, resolving

File: app/plugin_hooks_test.go (hunk 1 above)
Comment (arshchimni) [reply app/plugin_hooks_test.go]: Sure @mgdelacroix I'll update the pr

**Comment by arshchimni** [comment]
@mgdelacroix Made the changes that you requested. Nice catch! :)

**Pull request merged**